### PR TITLE
add editorconfig

### PR DESCRIPTION
**Pull request by OmgImAlexis**
Taken from https://github.com/jquery/api.qunitjs.com/blob/master/.editorconfig as per https://github.com/jquery/api.jqueryui.com/pull/293#issuecomment-163563691 suggested.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+root = true
+
+[*]
+indent_style = tab
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{json,yml}]
```
Comment (scottgonzalez) [inline .editorconfig]: The only `.json` file that uses spaces is `package.json`. Every other file uses tabs.

**Comment by scottgonzalez** [comment]
When adding a file like this, you should attempt to apply the rules to the existing codebase. That will give you an idea if the rules you're putting in place actually make sense for the project.

**Comment by scottgonzalez** [comment]
Why did you close all of your PRs?

**Comment by OmgImAlexis** [comment]
@scottgonzalez I'll just wait till you guys have your repos worked out. There's no point in keeping them open when I can't even merge them as you guys haven't added an editorconfig or anything similar to keep whitespace setup correctly. I would have figured from an organisation as big as jQuery these things would have been worked out years ago.

**Pull request closed without merging**